lib/new_relic/agent.rb

```diff
@@ -163,6 +170,86 @@ def add_or_defer_method_tracer(receiver, method_name, metric_name, options)
       end
     end
 
+    def self.add_automatic_method_tracers(arr)
+      return unless arr
+      return arr if arr.respond_to?(:empty?) && arr.empty?
+
+      arr = arr.split(/\s*,\s*/) if arr.is_a?(String)
+
+      add_tracers_once_methods_are_defined(arr.dup)
+
+      arr
+    end
+
+    # spawn a thread that will attempt to establish a tracer for each of the
+    # configured methods. the thread will continue to keep trying with each
+    # tracer until one of the following happens:
+    #   - the tracer is successfully established
+    #   - the configured method string couldn't be parsed
+    #   - establishing a tracer for a successfully parsed string failed
+    #   - the maximum number of attempts has been reached
+    # the thread will only be spawned once per agent initialization, to account
+    # for configuration reloading scenarios.
```

Comment:
I think these comments will show up on Rdoc. What do you think about adding an `# @api private` line to the end?

Reply:
Great call here. Added that for all new methods which are truly intended to be private here. Fixed in [87589d7](https://github.com/newrelic/newrelic-ruby-agent/pull/2851/commits/87589d7fe08f8c00c2e1a357b7d3e8a843d1caf6)